src/ObjectUtils/ElfFile.cpp

```diff
@@ -563,11 +602,15 @@ ErrorMessageOr<std::unique_ptr<ElfFile>> CreateElfFile(
 
   // Create appropriate ElfFile implementation
   if (llvm::dyn_cast<llvm::object::ELF32LEObjectFile>(object_file) != nullptr) {
-    result = std::unique_ptr<ElfFile>(
-        new ElfFileImpl<llvm::object::ELF32LE>(file_path, std::move(file)));
+    auto instance =
+        std::make_unique<ElfFileImpl<llvm::object::ELF32LE>>(file_path, std::move(file));
+    OUTCOME_TRY(instance->Initialize());
+    result = std::move(instance);
   } else if (llvm::dyn_cast<llvm::object::ELF64LEObjectFile>(object_file) != nullptr) {
-    result = std::unique_ptr<ElfFile>(
-        new ElfFileImpl<llvm::object::ELF64LE>(file_path, std::move(file)));
+    auto instance =
+        std::make_unique<ElfFileImpl<llvm::object::ELF64LE>>(file_path, std::move(file));
+    OUTCOME_TRY(instance->Initialize());
+    result = std::move(instance);
```

Comment:
I assume they code was different at some point, but now I don't see a reason for having `result`. Why not return early in each of these cases?

Reply:
makes sense. done.